src/txn/txn.c

```diff
@@ -1768,9 +1853,16 @@ __wt_txn_commit(WT_SESSION_IMPL *session, const char *cfg[])
         case WT_TXN_OP_BASIC_ROW:
         case WT_TXN_OP_INMEM_COL:
         case WT_TXN_OP_INMEM_ROW:
-            upd = op->u.op_upd;
-
             if (!prepare) {
+                /*
+                 * For now just try to resolve and if successful and returned an active hazard
+                 * reference, immediately release the active hazard pointer.
+                 */
+                WT_ERR(__txn_resolve_weak_hazard_updates(session, op, &ref));
```

Comment:
It is not okay to fail, see the comment above on line 1843:
"Note: we're going to commit: nothing can fail after this point."

prepare handles this by crashing the system. Do something similar - also check if that is the case with rollback too.

Reply:
Asserting for now if we do not resolve an update. Needs more work around this area.